### DISCUSSION TITLE
Unable to getAutoRenewProductId() on Server Notification event listeners

### DISCUSSION THREAD

**Discussion by r0hitch0pra** [imdhemy/laravel-in-app-purchases#91] [Q&A]
I have event listeners for almost all Apple IAP Server to server events. However, I don't understand how I am supposed to get "getAutoRenewProductId()" or even "getUnifiedReceipt()" from the event listener.

I have a need to store the renewal product ID into a database upon the DID_CHANGE_RENEWAL_PREF notification type. I can see that the event itself has all the properties as private/protected but neither getServerNotification() nor getSubscription() provides access to those functions/properties.

Is this intended or an oversight? Or am I doing something wrong?

**Comment by imdhemy** [comment]
@r0hitch0pra 
From the event, you can get the notification, from which you can get the subscription, and finally, you can use `getProviderRepresentation()` to get ios data.
